core/src/edu/asu/c3simulator/widgets/NavigationPanel.java

```diff
@@ -113,28 +146,12 @@ public void addButton(String buttonName, Screen targetedScreen)
 	 *            The screen that the button maps to. If null, no actionListener will be
 	 *            added to the button.
 	 */
-	public boolean addSubButton(String parentButtonName, String buttonName,
+	public void addSubButton(String parentButtonName, String buttonName,
 			Screen targetedScreen)
 	{
-		String storedParentButtonName = storedButtonName(parentButtonName);
-		if (!storedParentButtonName.equalsIgnoreCase(""))
-		{
-			if (subButtons.containsKey(storedParentButtonName))
-			{
-				LinkedHashMap<String, Screen> relativeSubButtons = subButtons
-						.get(storedParentButtonName);
-				relativeSubButtons.put(buttonName, targetedScreen);
-			}
-			else
-			{
-				LinkedHashMap<String, Screen> newSubButtons = new LinkedHashMap<String, Screen>();
-				newSubButtons.put(buttonName, targetedScreen);
-				subButtons.put(storedParentButtonName, newSubButtons);
-			}
-			recreateNavigationPanel();
-			return true;
-		}
-		return false;
+		ButtonNodeListener parent = buttons.get(parentButtonName);
```

Comment:
I guess what I mean is this: does the hash map compare strings using (equal) or (==)?